src/main/drivers/serial_uart_stm32f30x.c

```diff
@@ -38,67 +41,264 @@
+            { DEFIO_TAG_E(PA3),  DEFIO_TAG_E(PA2) },
+            { DEFIO_TAG_E(PB4),  DEFIO_TAG_E(PB3) },
+            { DEFIO_TAG_E(PD6),  DEFIO_TAG_E(PD5) },
+        },
+        .rx = IO_TAG_NONE,
+        .tx = IO_TAG_NONE,
+        .rcc = RCC_APB1(USART2),
+        .af = GPIO_AF_7,
+        .irqn = USART2_IRQn,
+        .txPriority = NVIC_PRIO_SERIALUART2_TXDMA,
+        .rxPriority = NVIC_PRIO_SERIALUART2_RXDMA,
+    },
+
+    // USART3
+    // PB11,PB10
+    // PC11,PC10
+    // PD9,PD8
+    {
+        .dev = USART3,
+        .rxDMAChannel = UART3_RX_DMA,
+        .txDMAChannel = UART3_TX_DMA,
+        .pinPair = {
+            { DEFIO_TAG_E(PB11), DEFIO_TAG_E(PB10) },
+            { DEFIO_TAG_E(PC11),  DEFIO_TAG_E(PC10) },
+            { DEFIO_TAG_E(PD9),  DEFIO_TAG_E(PD8) },
+        },
+        .rx = IO_TAG_NONE,
+        .tx = IO_TAG_NONE,
+        .rcc = RCC_APB1(USART3),
+        .af = GPIO_AF_7,
+        .irqn = USART3_IRQn,
+        .txPriority = NVIC_PRIO_SERIALUART3_TXDMA,
+        .rxPriority = NVIC_PRIO_SERIALUART3_RXDMA,
+    },
+
+    // UART4 XXX Not tested (yet!?) Need 303RC, e.g. LUX for testing
+    // PC11,PC10
+    {
+        .dev = UART4,
+        .rxDMAChannel = UART4_RX_DMA,
+        .txDMAChannel = UART4_TX_DMA,
+        .pinPair = {
+            { DEFIO_TAG_E(PC11), DEFIO_TAG_E(PC10) },
+        },
+        .rx = IO_TAG_NONE,
+        .tx = IO_TAG_NONE,
+        .rcc = RCC_APB1(UART4),
+        .af = GPIO_AF_5,
+        .irqn = UART4_IRQn,
+        .txPriority = NVIC_PRIO_SERIALUART4_TXDMA,
+        .rxPriority = NVIC_PRIO_SERIALUART4_RXDMA,
+    },
+
+    // UART5 XXX Not tested (yet!?) Need 303RC; e.g. LUX for testing
+    // PD2,PC12
+    {
+        .dev = UART5,
+        .rxDMAChannel = 0,
+        .txDMAChannel = 0,
+        .pinPair = {
+            { DEFIO_TAG_E(PD2), DEFIO_TAG_E(PC12) },
+        },
+        .rx = IO_TAG_NONE,
+        .tx = IO_TAG_NONE,
+        .rcc = RCC_APB1(UART5),
+        .af = GPIO_AF_5,
+        .irqn = UART5_IRQn,
+        .txPriority = NVIC_PRIO_SERIALUART5,
+        .rxPriority = NVIC_PRIO_SERIALUART5,
+    },
+};
+
+static uartDevice_t *uartHardwareMap[ARRAYLEN(uartHardware)];
+
+// XXX What a waste...
+// XXX These could be left inside the serialUARTx() function,
+// XXX but they will eventually go away...
+// XXX ARRAYLEN(uartHardware) is overkill; What was the number (count) of UART serial ports???
+
+static volatile uint8_t rxBuffers[ARRAYLEN(uartHardware)][UART_RX_BUFFER_SIZE];
+static volatile uint8_t txBuffers[ARRAYLEN(uartHardware)][UART_TX_BUFFER_SIZE];
+
+void serialInitHardwareMap(serialPinConfig_t *pSerialPinConfig)
+{
+    memset(uartHardwareMap, 0, sizeof(uartHardwareMap));
+
+    for (size_t index = 0 ; index < ARRAYLEN(uartHardware) ; index++) {
+        uartDevice_t *uartDev = &uartHardware[index];
+        for (int pair = 0 ; pair < UART_PINPAIR_COUNT ; pair++) {
+
+            if (uartDev->pinPair[pair].rx == pSerialPinConfig->ioTagRx[index]
```

Comment:
Maybe this test may be relaxed - BIDIR or RX/TX only port does not need the second pin

Another useful feature is to be able to remap pins when sharing port - attach both devices to different pins and remap UART on arm. Pin pair will have to be parameter of serialOpen, which will need some refactoring ...